### PR TITLE
Hexagon shading improvement (Battlefield)

### PR DESCRIPTION
**Pull request by Districh-ru**
In addition to #6216: made generation of two hexagon shadows.
- shadow that fits the grid
изображение

-  shadow with 3px space between hexagons for use without grid
изображение

(Cursor shadow size is set to fit the grid.)

### CODE REVIEW

**Review by Branikolog — APPROVED** [review]

File: src/fheroes2/battle/battle_interface.cpp [hunk 1]
```diff
@@ -499,15 +499,15 @@ fheroes2::Image DrawHexagon( const uint8_t colorId )
     return sf;
 }
 
-fheroes2::Image DrawHexagonShadow( const uint8_t alphaValue )
+fheroes2::Image DrawHexagonShadow( const uint8_t alphaValue, const uint8_t horizSpace )
```
Comment (ihhub) [inline src/fheroes2/battle/battle_interface.cpp:502]: `horizSpace` variable should be `int32_t` to avoid implicit type conversions.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @Districh-ru , I added few comments here. Would you mind to take a look at them?

**Comment by Districh-ru** [comment]
Hi @ihhub, I've updated the code.
Thanks for the optimization suggestions!

**Review by ihhub — APPROVED** [review]
@Districh-ru , now the code is clean and nice to read. Well done!

**Comment by ihhub** [comment]
@Districh-ru , thank you so much for this improvement!

**Pull request merged**